examples/obstacle_circumnavigation/simulated_obstacle_avoiding_positioner.cc

```diff
@@ -80,34 +81,43 @@ int bobMain(int, char **)
 
     bool runPositioner = false;
     while (display.isOpen()) {
-        // Run GUI events
         car.setPose(robot.getPose());
-        const sf::Event event = display.updateAndDrive(robot, objectShape, goalCircle, car);
 
-        // Spacebar toggles whether positioner is running
-        if (event.type == sf::Event::KeyReleased && event.key.code == sf::Keyboard::Space) {
-            runPositioner = !runPositioner;
-            robot.stopMoving();
-        }
+        auto eventHandler = [&](const sf::Event &event) {
+            // drive robot with keyboard
+            if (Viz::SFML::drive(robot, event)) {
+                return;
+            }
 
-        if (runPositioner) {
-            if (display.mouseClicked()) {
-                // Set a new goal position if user clicks in the window
-                const auto mousePosition = display.mouseClickPosition();
+            // Spacebar toggles whether positioner is running
+            if (event.type == sf::Event::KeyReleased && event.key.code == sf::Keyboard::Space) {
+                runPositioner = !runPositioner;
+                robot.stopMoving();
+                return;
+            }
+
+            if (runPositioner && event.type == sf::Event::MouseButtonPressed
+                && event.mouseButton.button == sf::Mouse::Left) {
+
+                // set a new goal position if user clicks in the window
+                const auto vec = display.pixelToVector(event.mouseButton.x, event.mouseButton.y);
 
                 // Set the goal to this position
-                const Pose2<meter_t, radian_t> goalPose{ mousePosition.x(), mousePosition.y(), 15_deg };
+                const Pose2<meter_t, radian_t> goalPose{ vec.x(), vec.y(), 15_deg };
```

Comment:
is this an arbitrary angle?

Reply:
Yeah. I didn't want it to be exactly 0° because it's harder to tell if things are broken that way

Reply:
makes sense, might be worth a comment though